File: apt-integration-tests/src/main/java/me/kpavlov/kt/schema/apt/integration/type/Status.java
package me.kpavlov.kt.schema.apt.integration.type;

import com.fasterxml.jackson.annotation.JsonClassDescription;

/**
 * Simple test model to verify Java enum schema generation.
 */
@JsonClassDescription("Current lifecycle status of an entity.")
public enum Status {
    ACTIVE, INACTIVE, PENDING
}

File: ksp-integration-tests/src/main/kotlin/me/kpavlov/kt/schema/integration/type/Status.kt
package me.kpavlov.kt.schema.integration.type

import me.kpavlov.kt.schema.Description
import me.kpavlov.kt.schema.Schema

/**
 * Enum class with Schema annotation
 */
@Description("Current lifecycle status of an entity.")
@Schema
enum class Status {
    @Description("Entity is active and usable")
    ACTIVE,

    @Description("Entity is inactive or disabled")
    INACTIVE,

    @Description("Entity is pending activation or approval")
    PENDING,
}
